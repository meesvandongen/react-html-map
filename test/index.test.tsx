import * as React from 'react';
import { render } from '@testing-library/react';
import HtmlMapper from '../src';

const TEST_HTML =
    '<p id="test-html">Af deel pomp soms tijd veel ad. En <strong>voorloopig</strong> uitgegeven en nu ad verkochten beschikken. Al zout al in over bord te. Voorschijn moeilijker wedijveren na op. Zelf kilo zoon wel dag ruwe gas. Grayah de op vloeit na is goeden. Cenis langs maken nemen al ad klein de te.<br />Bepaalde gebruikt de verrezen <em>gestoken schatten</em> en <strong>verbouwd</strong>. <br />Krachten nu eveneens na in op nadering britsche maleiers verbruik. </p>\\n<p>Dat geheel vleesch zonder been is, gebruikt men om daar Osse-worst van te maaken; en het vleesch van de schouwders met de twee platte bil stukken, met het vet van de broek gebruikt men tot rolpens in plaats van ander vet.</p>\\n<ul><li>Dit is een feest</li><li>Morgen komt de melkboer</li></ul><h3>Al zout al in over bord te.</h3>\\n<p>Witheid meestal noemden met zee aandeel gezocht valorem heb. Holen moest steek zoo mei zit. Slechts zee dag bronnen gemengd weg behoeft doelang der. Al blijft midden op om na daarin. Dien werk van eind vaak zal per doel iets gif. Tembun wat groote een enkele.</p>\\n<h3>Lauriergracht No 37</h3>\\n<p>Ik ben makelaar in koffi, en woon op de Lauriergracht No 37. Het is mijn gewoonte niet, romans te schrijven, of zulke dingen, en het heeft dan ook lang geduurd, voor ik er toe overging een paar riem papier extra te bestellen, en het werk aan te vangen, dat gij, lieve lezer, zoâven in de hand hebt genomen, en dat ge lezen moet als ge makelaar in koffie zijt, of als ge wat anders zijt.</p>\\n<h4>Busselinck &amp; Waterman</h4>\\n<p>Dat zijn ook makelaars in koffie, doch hun adres behoeft ge niet te weten. Ik pas er dus wel op, dat ik geen romans schrijf, of andere valse opgaven doe. Ik heb dan ook altijd opgemerkt dat mensen die zich met zoiets inlaten, gewoonlijk slecht wegkomen. Ik ben drieënveertig jaar oud, bezoek sedert twintig jaren de beurs, en kan dus voor de dag treden, als men iemand roept die ondervinding heeft. Ik heb al wat huizen zien vallen!</p>';

describe('HtmlMapper', () => {
    it('renders whitelisted components', () => {
        const { container } = render(
            <HtmlMapper html={TEST_HTML}>
                {{
                    h3: null,
                    p: null,
                }}
            </HtmlMapper>
        );
        expect(container.firstChild).toMatchSnapshot();
    });
    
    it('renders all non-whitelisted components with acceptUnknown enabled', () => {
        const { container } = render(
            <HtmlMapper html={TEST_HTML} acceptUnknown>
                {{}}
            </HtmlMapper>
        );
        expect(container.firstChild).toMatchSnapshot();
    });
    
    it('renders custom components', () => {
        const { container } = render(
            <HtmlMapper html={TEST_HTML}>
                {{
                    p: ({ id, children }) => (
                        // a different html tag, and a attribute mapped to a different attribute.
                        <h6 className={id}>{children}</h6>
                    ),
                }}
            </HtmlMapper>
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
