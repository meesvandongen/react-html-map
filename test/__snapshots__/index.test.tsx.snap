// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HtmlMapper > handles HTML entities correctly 1`] = `
<div>
  <p>
    & &lt; &gt; "
  </p>
</div>
`;

exports[`HtmlMapper > handles HTML entities with decodeEntities disabled 1`] = `
<div>
  <p>
    &amp; &lt; &gt; &quot;
  </p>
</div>
`;

exports[`HtmlMapper > handles elements with no children 1`] = `
<div>
  <div />
  <p />
</div>
`;

exports[`HtmlMapper > handles empty tag names by rendering fragments 1`] = `<div />`;

exports[`HtmlMapper > handles mixed content with custom renderer that returns null 1`] = `
<div>
  <div>
    Text 
     more text
  </div>
</div>
`;

exports[`HtmlMapper > handles nodes that are not Text or Element (covers return null case) 1`] = `
<div>
  <div />
</div>
`;

exports[`HtmlMapper > handles text nodes correctly 1`] = `
<div>
  Just plain text
</div>
`;

exports[`HtmlMapper > handles unknown elements without acceptUnknown 1`] = `
<div>
  <div />
</div>
`;

exports[`HtmlMapper > handles void elements correctly 1`] = `
<div>
  <div>
    <br />
    <hr />
    <img
      alt="test"
      src="test.jpg"
    />
    <input
      type="text"
    />
  </div>
</div>
`;

exports[`HtmlMapper > passes index parameter to custom renderers 1`] = `
<div>
  <div>
    <p
      class="first-paragraph"
      data-index="0"
    >
      First
    </p>
    <p
      class="paragraph"
      data-index="1"
    >
      Second
    </p>
    <p
      class="paragraph"
      data-index="2"
    >
      Third
    </p>
  </div>
</div>
`;

exports[`HtmlMapper > renders all non-whitelisted components with acceptUnknown enabled 1`] = `
<div>
  <p
    id="test-html"
  >
    Af deel pomp soms tijd veel ad. En 
    <strong>
      voorloopig
    </strong>
     uitgegeven en nu ad verkochten beschikken. Al zout al in over bord te. Voorschijn moeilijker wedijveren na op. Zelf kilo zoon wel dag ruwe gas. Grayah de op vloeit na is goeden. Cenis langs maken nemen al ad klein de te.
    <br />
    Bepaalde gebruikt de verrezen 
    <em>
      gestoken schatten
    </em>
     en 
    <strong>
      verbouwd
    </strong>
    . 
    <br />
    Krachten nu eveneens na in op nadering britsche maleiers verbruik. 
  </p>
  \\n
  <p>
    Dat geheel vleesch zonder been is, gebruikt men om daar Osse-worst van te maaken; en het vleesch van de schouwders met de twee platte bil stukken, met het vet van de broek gebruikt men tot rolpens in plaats van ander vet.
  </p>
  \\n
  <ul>
    <li>
      Dit is een feest
    </li>
    <li>
      Morgen komt de melkboer
    </li>
  </ul>
  <h3>
    Al zout al in over bord te.
  </h3>
  \\n
  <p>
    Witheid meestal noemden met zee aandeel gezocht valorem heb. Holen moest steek zoo mei zit. Slechts zee dag bronnen gemengd weg behoeft doelang der. Al blijft midden op om na daarin. Dien werk van eind vaak zal per doel iets gif. Tembun wat groote een enkele.
  </p>
  \\n
  <h3>
    Lauriergracht No 37
  </h3>
  \\n
  <p>
    Ik ben makelaar in koffi, en woon op de Lauriergracht No 37. Het is mijn gewoonte niet, romans te schrijven, of zulke dingen, en het heeft dan ook lang geduurd, voor ik er toe overging een paar riem papier extra te bestellen, en het werk aan te vangen, dat gij, lieve lezer, zoâven in de hand hebt genomen, en dat ge lezen moet als ge makelaar in koffie zijt, of als ge wat anders zijt.
  </p>
  \\n
  <h4>
    Busselinck & Waterman
  </h4>
  \\n
  <p>
    Dat zijn ook makelaars in koffie, doch hun adres behoeft ge niet te weten. Ik pas er dus wel op, dat ik geen romans schrijf, of andere valse opgaven doe. Ik heb dan ook altijd opgemerkt dat mensen die zich met zoiets inlaten, gewoonlijk slecht wegkomen. Ik ben drieënveertig jaar oud, bezoek sedert twintig jaren de beurs, en kan dus voor de dag treden, als men iemand roept die ondervinding heeft. Ik heb al wat huizen zien vallen!
  </p>
</div>
`;

exports[`HtmlMapper > renders custom components 1`] = `
<div>
  <h6
    class="test-html"
  >
    Af deel pomp soms tijd veel ad. En 
     uitgegeven en nu ad verkochten beschikken. Al zout al in over bord te. Voorschijn moeilijker wedijveren na op. Zelf kilo zoon wel dag ruwe gas. Grayah de op vloeit na is goeden. Cenis langs maken nemen al ad klein de te.
    Bepaalde gebruikt de verrezen 
     en 
    . 
    Krachten nu eveneens na in op nadering britsche maleiers verbruik. 
  </h6>
  \\n
  <h6>
    Dat geheel vleesch zonder been is, gebruikt men om daar Osse-worst van te maaken; en het vleesch van de schouwders met de twee platte bil stukken, met het vet van de broek gebruikt men tot rolpens in plaats van ander vet.
  </h6>
  \\n
  \\n
  <h6>
    Witheid meestal noemden met zee aandeel gezocht valorem heb. Holen moest steek zoo mei zit. Slechts zee dag bronnen gemengd weg behoeft doelang der. Al blijft midden op om na daarin. Dien werk van eind vaak zal per doel iets gif. Tembun wat groote een enkele.
  </h6>
  \\n
  \\n
  <h6>
    Ik ben makelaar in koffi, en woon op de Lauriergracht No 37. Het is mijn gewoonte niet, romans te schrijven, of zulke dingen, en het heeft dan ook lang geduurd, voor ik er toe overging een paar riem papier extra te bestellen, en het werk aan te vangen, dat gij, lieve lezer, zoâven in de hand hebt genomen, en dat ge lezen moet als ge makelaar in koffie zijt, of als ge wat anders zijt.
  </h6>
  \\n
  \\n
  <h6>
    Dat zijn ook makelaars in koffie, doch hun adres behoeft ge niet te weten. Ik pas er dus wel op, dat ik geen romans schrijf, of andere valse opgaven doe. Ik heb dan ook altijd opgemerkt dat mensen die zich met zoiets inlaten, gewoonlijk slecht wegkomen. Ik ben drieënveertig jaar oud, bezoek sedert twintig jaren de beurs, en kan dus voor de dag treden, als men iemand roept die ondervinding heeft. Ik heb al wat huizen zien vallen!
  </h6>
</div>
`;

exports[`HtmlMapper > renders fragment when name is falsy (covers Fragment return) 1`] = `
<div>
  <div
    data-testid="wrapper"
  >
    test
  </div>
</div>
`;

exports[`HtmlMapper > renders whitelisted components 1`] = `
<div>
  <p
    id="test-html"
  >
    Af deel pomp soms tijd veel ad. En 
     uitgegeven en nu ad verkochten beschikken. Al zout al in over bord te. Voorschijn moeilijker wedijveren na op. Zelf kilo zoon wel dag ruwe gas. Grayah de op vloeit na is goeden. Cenis langs maken nemen al ad klein de te.
    Bepaalde gebruikt de verrezen 
     en 
    . 
    Krachten nu eveneens na in op nadering britsche maleiers verbruik. 
  </p>
  \\n
  <p>
    Dat geheel vleesch zonder been is, gebruikt men om daar Osse-worst van te maaken; en het vleesch van de schouwders met de twee platte bil stukken, met het vet van de broek gebruikt men tot rolpens in plaats van ander vet.
  </p>
  \\n
  <h3>
    Al zout al in over bord te.
  </h3>
  \\n
  <p>
    Witheid meestal noemden met zee aandeel gezocht valorem heb. Holen moest steek zoo mei zit. Slechts zee dag bronnen gemengd weg behoeft doelang der. Al blijft midden op om na daarin. Dien werk van eind vaak zal per doel iets gif. Tembun wat groote een enkele.
  </p>
  \\n
  <h3>
    Lauriergracht No 37
  </h3>
  \\n
  <p>
    Ik ben makelaar in koffi, en woon op de Lauriergracht No 37. Het is mijn gewoonte niet, romans te schrijven, of zulke dingen, en het heeft dan ook lang geduurd, voor ik er toe overging een paar riem papier extra te bestellen, en het werk aan te vangen, dat gij, lieve lezer, zoâven in de hand hebt genomen, en dat ge lezen moet als ge makelaar in koffie zijt, of als ge wat anders zijt.
  </p>
  \\n
  \\n
  <p>
    Dat zijn ook makelaars in koffie, doch hun adres behoeft ge niet te weten. Ik pas er dus wel op, dat ik geen romans schrijf, of andere valse opgaven doe. Ik heb dan ook altijd opgemerkt dat mensen die zich met zoiets inlaten, gewoonlijk slecht wegkomen. Ik ben drieënveertig jaar oud, bezoek sedert twintig jaren de beurs, en kan dus voor de dag treden, als men iemand roept die ondervinding heeft. Ik heb al wat huizen zien vallen!
  </p>
</div>
`;

exports[`HtmlMapper > supports tag mapping with null for pass-through rendering 1`] = `
<div>
  <div>
    <p>
      Paragraph
    </p>
    <strong>
      Bold
    </strong>
  </div>
</div>
`;
